client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -291,8 +291,19 @@ public void checkServerTrusted(X509Certificate[] x509Certificates, String s) thr
                 if (hostnameVerifier != null || sslSocketFactory != null || sslContext != null) {
                 	TLSClientParameters tlsClientParameters = new TLSClientParameters();
                 	tlsClientParameters.setHostnameVerifier(hostnameVerifier);
-                	tlsClientParameters.setSSLSocketFactory(sslSocketFactory);
-                	tlsClientParameters.setSslContext(sslContext);
+                	if(sslSocketFactory != null) {
+                    	if(sslSocketFactory.getDefaultCipherSuites()!=null && Arrays.asList(sslSocketFactory.getDefaultCipherSuites()).size()>0) {
+                    		tlsClientParameters.setCipherSuites(Arrays.asList(sslSocketFactory.getDefaultCipherSuites()));
```

Comment:
Looking at how the `tlsClientParameters.getCipherSuites` is used, I'm not sure how this will get used:
* `HttpConduitConfigApplier.applyTlsClientParameters` has no effect (code only used in `else` block where SSLSocketFactory was null)
* `HttpConduitConfigApplier.applyTlsClientParameters` looks worrying: it adds to the list by calling `p.getCipherSuites().add(...)`. That code is private so hopefully never called with the cipher suite you pass in - if it was, it would fail (because `Arrays.asList` returns a fixed-size list.
* `AsyncHTTPConduit.initializeSSLEngine`: not sure what calls this.